### ISSUE TITLE
Investigate calculation inaccuracy

### ISSUE THREAD

**Issue by LinhWuerzburger** [FireDynamics/ARTSS#89]
There are several occasions on which a value is multiplied by the inverse of a number instead of a division. This may yield to inaccuracy, and can be easly fixed by changing the calculation order, but each occasion has to be evaluated individually.

**Comment by ChristianVonMach** [comment]
found a lot of code matching the description.
found occurrence:

- `boundary/Surface.cpp`:22-24
  - `real rdx = 1./dx;`
  - ...
- `src/diffusion/JacobiDiffuse.cpp`:64-66,73, (pattern repeats itself in 179-181,379-381,405) and in `src/diffusion/ColoredGaussSeidelDiffuse.cpp`
  - `const real rdx = 1. / dx;`
  - ...
  - `const real beta = 1. / rbeta;`
 - `diffusion/ColoredGaussSeidelDiffuse.cpp`:381,393,408,...
   - calculation `1. / bb` passing as an parameter to `ColoredGaussSeidelDiffuse::colored_gauss_seidel_stencil`  but in this function `1. / bb` is used as an factor.
 
(to be continued)

**Comment by ChristianVonMach** [comment]
But i would like to point out multiplication is much faster than division.
https://stackoverflow.com/a/2860396

**Comment by LinhWuerzburger** [comment]
Yes, multiplication is faster and that is the reason why it was implemented that way. But at some points, it creates problematic inaccuracy. For example, if you have `reverse_x = 1/x; x * reverse_x != 1` it is quite a big issue and we can't choose performance over accuracy in this case. So I am not saying to replace it everywhere, but everywhere where it creates inaccuracies. There was an example... though I have to admit I forgot where I changed it.

**Comment by ChristianVonMach** [comment]
since the numerical error is determined by |log(|x|)| i guess there is no definitive answer other then it depends (on the input case). Calculating the numerical error is hard and unfeasible.
